odl/trafos/wavelet.py

```diff
@@ -606,23 +912,47 @@ def __init__(self, dom, nscales, wbasis, mode):
             raise ValueError('Cannot use more than {} scaling levels, '
                              'got {}.'.format(max_level, self.nscales))
 
-        self.size_list = coeff_size_list(dom.shape, self.nscales,
-                                         self.wbasis, self.mode)
-
-        ran_size = np.prod(self.size_list[0])
-
-        if dom.ndim == 1:
-            ran_size += sum(np.prod(shape) for shape in
-                            self.size_list[1:-1])
-        elif dom.ndim == 2:
-            ran_size += sum(3 * np.prod(shape) for shape in
-                            self.size_list[1:-1])
-        elif dom.ndim == 3:
-            ran_size += sum(7 * np.prod(shape) for shape in
-                            self.size_list[1:-1])
+        self._axes = axes
+        if axes is not None:
+            if len(axes) != dom.ndim:
+                raise ValueError('The length of the axes has to be equal to '
+                                 'dimension. Axes len is {} and dimension '
+                                 'is {}.'.format(len(axes), dom.ndim))
+            else:
+                # TODO new helper function for the size_list with axes option
+                self.size_list = coeff_size_list_axes(dom.shape, self.wbasis,
+                                                      self.mode, self._axes)
+                ran_size = np.prod(self.size_list[0])
+
+                if dom.ndim == 1:
+                    ran_size += sum(np.prod(shape) for shape in
+                                    self.size_list[1:-1])
+                elif dom.ndim == 2:
+                    ran_size += sum(3 * np.prod(shape) for shape in
+                                    self.size_list[1:-1])
+                elif dom.ndim == 3:
+                    ran_size += sum(7 * np.prod(shape) for shape in
+                                    self.size_list[1:-1])
+                else:
+                    raise NotImplementedError('ndim {} not 1, 2 or 3'
+                                              ''.format(len(dom.ndim)))
         else:
-            raise NotImplementedError('ndim {} not 1, 2 or 3'
-                                      ''.format(len(dom.ndim)))
+            self.size_list = coeff_size_list(dom.shape, self.nscales,
+                                             self.wbasis, self.mode)
+
+            ran_size = np.prod(self.size_list[0])
+            if dom.ndim == 1:
+                ran_size += sum(np.prod(shape) for shape in
+                                self.size_list[1:-1])
+            elif dom.ndim == 2:
+                ran_size += sum(3 * np.prod(shape) for shape in
+                                self.size_list[1:-1])
+            elif dom.ndim == 3:
+                ran_size += sum(7 * np.prod(shape) for shape in
+                                self.size_list[1:-1])
```

Comment:
This is a good candidate for a helper function, like `flat_size_from_size_list`.